store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreGraphIDsTest.java

```diff
@@ -0,0 +1,248 @@
+                .build();
+        final Entity thirdEntity = new Entity.Builder()
+                .group(GROUP_BASIC_ENTITY)
+                .vertex(SOURCE_BASIC)
+                .property(PROPERTY_1, 3)
+                .build();
+
+        federatedStore.execute(
+                new AddElements.Builder()
+                        .input(secondEntity)
+                        .option(DEPRECATED_GRAPHIDS_OPTION, GRAPH_ID_A)
+                        .build(), contextBlankUser());
+        federatedStore.execute(
+                new AddElements.Builder()
+                        .input(thirdEntity)
+                        .option(DEPRECATED_GRAPHIDS_OPTION, GRAPH_ID_B)
+                        .build(), contextBlankUser());
+
+        // When
+        Iterable<? extends Element> results = (Iterable<? extends Element>) federatedStore.execute(
+                new OperationChain.Builder()
+                        .first(new GetElements.Builder()
+                                .input(BASIC_VERTEX)
+                                .option(DEPRECATED_GRAPHIDS_OPTION, GRAPH_ID_A)
+                                .build())
+                        .then(new ExportToSet.Builder<>()
+                                .key("results")
+                                .build())
+                        .then(new DiscardOutput())
+                        .then(new GetElements.Builder()
+                                .input(SOURCE_BASIC)
+                                .option(DEPRECATED_GRAPHIDS_OPTION, GRAPH_ID_B)
+                                .build())
+                        .then(new ExportToSet.Builder<>()
+                                .key("results")
+                                .build())
+                        .then(new DiscardOutput())
+                        .then(new GetSetExport.Builder()
+                                .key("results")
+                                .build())
+                        .build(), contextBlankUser());
+
+        // Then
+        assertThat(results).hasSize(2);
+        assertThat((Iterable<Entity>) results).containsExactlyInAnyOrder(EXPECTED_ENTITY, thirdEntity);
+    }
+
+    @Test
+    public void shouldNotCorrectlySelectGraphUsingGraphIdsOptionOnOperationChain() throws Exception {
+        // When
+        Iterable<? extends Element> results = federatedStore.execute(
+                new OperationChain.Builder()
+                        .first(new GetElements.Builder()
+                                .input(EXPECTED_ENTITY)
+                                .build())
+                        .then(new Limit<>(10))
+                        .option(DEPRECATED_GRAPHIDS_OPTION, GRAPH_ID_A)
+                        .build(), contextBlankUser());
+
+        // Then
+        // Pre Gaffer 2.0 this would have sent the whole chain to graph A
+        // Now, the chain's graphIds option is ignored as every individual
+        // operation is wrapped in a FederatedOperation
```

Comment:
So the addition of the option doesn't have any effect here because it is attached to the chain and not to the individual operation inside it. The important bit here being that `DEPRECATED_GRAPHIDS_OPTION` only has any effect when used on individual operations and not on a chain?
Could have some more info on that, seeing as there is a comment.

Reply:
Needs to be documented for sure